crates/uv-virtualenv/src/virtualenv.rs

```diff
@@ -120,6 +121,11 @@ pub(crate) fn create(
                     debug!("Allowing existing {name} due to `--allow-existing`");
                 }
                 OnExisting::Remove => {
+                    // Refuse to remove a non-virtual environment, even if `--clear` is provided.
+                    if !is_virtualenv {
+                        return Err(Error::NonVirtualEnvironment(location.to_path_buf()));
+                    }
```

Comment:
This is the part I would consider moving into a breaking release / requiring a `--force` flag.